transport/tchannel/config.go

```diff
@@ -61,7 +61,16 @@ type OutboundConfig struct {
 
 // TransportSpec returns a TransportSpec for the TChannel unary transport.
 func TransportSpec(opts ...Option) config.TransportSpec {
-	return (&transportSpec{}).Spec(opts...)
+	var ts transportSpec
+	for _, o := range opts {
+		switch opt := o.(type) {
+		case TransportOption:
```

Comment:
Possible to prevent configuring the address with an option?

I think it will be much clearer if config is the only way to configure the address, e.g. no options.